app/forms/questionnaire_form.py

```diff
@@ -492,31 +458,26 @@ def get_answer_fields(
     question: QuestionSchemaType,
     data: MultiDict[str, Any] | Mapping[str, Any] | None,
     schema: QuestionnaireSchema,
-    answer_store: AnswerStore,
-    list_store: ListStore,
-    metadata: MetadataProxy | None,
-    response_metadata: MutableMapping,
+    data_stores: DataStores,
     location: LocationType | None,
-    progress_store: ProgressStore,
-    supplementary_data_store: SupplementaryDataStore,
 ) -> dict[str, FieldHandler]:
     list_item_id = location.list_item_id if location else None
 
     block_ids_by_section: dict[SectionKey, tuple[str, ...]] = {}
 
-    if location and progress_store:
+    if location and data_stores.progress_store:
         block_ids_by_section = (
             get_routing_path_block_ids_by_section_for_calculated_summary_dependencies(
                 location=location,
-                progress_store=progress_store,
+                progress_store=data_stores.progress_store,
                 path_finder=PathFinder(
                     schema=schema,
-                    answer_store=answer_store,
-                    list_store=list_store,
-                    progress_store=progress_store,
-                    metadata=metadata,
-                    response_metadata=response_metadata,
-                    supplementary_data_store=supplementary_data_store,
+                    answer_store=data_stores.answer_store,
+                    list_store=data_stores.list_store,
+                    progress_store=data_stores.progress_store,
+                    metadata=data_stores.metadata,
+                    response_metadata=data_stores.response_metadata,
+                    supplementary_data_store=data_stores.supplementary_data_store,
```

Comment:
It's because `PathFinder` is initialised in `PlaceholderParser` which is initialised in `PlaceholderRenderer` which is part of a helper method in `DataStores` class and we pass in separate stores as arguments at `DataStores` level. But we can recreate `DataStores` as one entity in parser to pass it to path finder.